src/geom/LineSegment.cpp

```diff
@@ -55,6 +55,9 @@ LineSegment::projectionFactor(const CoordinateXY& p) const
     if(p == p1) {
         return 1.0;
     }
+    if(p0 == p1) {
+        return 0.0;
+    }
```

Comment:
This is the case of projecting on a LineSegment where the start and end point is the same (which otherwise ends up doing a 0/0 in the code below)